mitodl/release-script
# Fix Doof reenterant bug, wait for deployment when Doof starts up

#### What are the relevant tickets?
Fixes #125. That is for a command `wait for deploy`, but I don't think a command like that would be necessary as long as we do this check each time doof starts up. 

Also fixes #212. I think the intermittent release failure has happened because there is an async context manager `init_working_dir` which creates a temp directory then switches the current directory to that directory. The problem happens when there are two `init_working_dir` context managers which are entered at the same time the current directory. In that case the current working directory will be wrong for one of them. This PR changes the current working directory to be passed in explicitly so that two coroutines can operate at the same time without interfering with each other.

#### What's this PR do?
 - when Doof starts up, check each channel and wait for deployment if necessary
 - prevent wait for checkboxes and wait for deploys from being run twice for the same channel
 - remove `wait_for_deploy.sh` and other related command line functionality which isn't being used

#### How should this be manually tested?
Merge, then make sure nothing has changed with doof. It will probably be difficult to tell if everything is working fine since it relies on a restart during the deployment process which is only 20 minutes or so at the longest.
